### PR TITLE
Add `Node Classification` tutorial GNNLux 

### PR DESCRIPTION
**Pull request by aurorarossi**
Translation of the `Node Classification` tutorial from GNNFlux

### CODE REVIEW

File: GNNLux/docs/src_tutorials/node_classification.jl [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+# Our MLP is defined by two linear layers and enhanced by [ReLU](https://lux.csail.mit.edu/stable/api/NN_Primitives/ActivationFunctions#NNlib.relu) non-linearity and [Dropout](https://lux.csail.mit.edu/stable/api/Lux/layers#Lux.Dropout).
+# Here, we first reduce the 1433-dimensional feature vector to a low-dimensional embedding (`hidden_channels=16`), while the second linear layer acts as a classifier that should map each low-dimensional node embedding to one of the 7 classes.
+
+# Let's train our simple MLP by following a similar procedure as described in [the first part of this tutorial](https://juliagraphs.org/GraphNeuralNetworks.jl/docs/GNNLux.jl/stable/tutorials/gnn_intro/).
+# We again make use of the **cross entropy loss** and **Adam optimizer**.
+# This time, we also define a **`accuracy` function** to evaluate how well our final model performs on the test node set (which labels have not been observed during training).
+
+
+function custom_loss(model, ps, st, x)
```
Comment (CarloLucibello) [inline GNNLux/docs/src_tutorials/node_classification.jl:99]:
```suggestion
function loss(model, ps, st, x)
```

File: GNNLux/docs/src_tutorials/node_classification.jl [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+# where $\mathbf{W}^{(\ell + 1)}$ denotes a trainable weight matrix of shape `[num_output_features, num_input_features]` and $c_{w,v}$ refers to a fixed normalization coefficient for each edge.
+# In contrast, a single `Linear` layer is defined as
+
+# ```math
+# \mathbf{x}_v^{(\ell + 1)} = \mathbf{W}^{(\ell + 1)} \mathbf{x}_v^{(\ell)}
+# ```
+
+# which does not make use of neighboring node information.
+
+Lux.@concrete struct GCN <: GNNContainerLayer{(:conv1, :drop, :conv2)}
```
Comment (CarloLucibello) [inline GNNLux/docs/src_tutorials/node_classification.jl:155]: `@concrete` is not part of Lux, it is from ConcreteStructs.jl

File: GNNLux/docs/src_tutorials/node_classification.jl [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+# function LuxCore.initialparameters(rng::TaskLocalRNG, l::GCN) # initialize model parameters
+#     weight_c1 = l.init_weight(rng, l.hd, l.nf)
+#     weight_c2 = l.init_weight(rng, l.nc, l.hd)
+#     if l.use_bias
+#         bias_c1 = l.init_bias(rng, l.hd)
+#         bias_c2 = l.init_bias(rng, l.nc)
+#         return (; conv1 = ( weight = weight_c1, bias = bias_c1),  drop= LuxCore.initialparameters(rng, l.drop), conv2 = ( weight = weight_c2, bias = bias_c2))
+#     end
+#     return (; conv1 = ( weight = weight_c1), drop= LuxCore.initialparameters(rng, l.drop),  conv2 = ( weight = weight_c2))
+# end
```
Comment (CarloLucibello) [inline GNNLux/docs/src_tutorials/node_classification.jl:192]: this can be eliminated, right?

Comment (aurorarossi) [reply GNNLux/docs/src_tutorials/node_classification.jl:192]: yes

**Comment by aurorarossi** [comment]
#545

**Pull request merged**